lib/assets/javascripts/views/users/admin_grant_permissions.js

```diff
@@ -0,0 +1,65 @@
+import { paginableSelector } from '../../utils/paginable';
+
+$(() => {
+  const success = (data) => {
+    // Render the html in the modal-permissions modal
+    $('#modal-permissions').html(data.user.html);
+    if ($('.org_grant_privileges:checked').length === $('.org_grant_privileges').length) {
+      $('#modal-permissions #org_admin_privileges').prop('checked', true);
+    }
+    if ($('.super_grant_privileges:checked').length === $('.super_grant_privileges').length) {
+      $('#modal-permissions #super_admin_privileges').prop('checked', true);
+    }
+  };
+
+  const error = () => {
+    // There was an ajax error so just route the user to the sign-in modal
+    // and let them sign in as a Non-Partner Institution
+    $('a[data-target="#modal-permissions"]').tab('show');
+  };
+
+  $(paginableSelector).on('click', '.modal-window', (e) => {
```

Comment:
We are using unobtrusive JavaScript across the site which means $.ajax is abstracted for us. I think is worth to be consistent for any new requirement coming. Can you refactor from L21-31 using delegated events for success and error. You can see a similar example at lib/assets/javascripts/views/plans/index.js